File: SuperiorPackGroup/BLL/LocationsBLL.vb
Imports DXDAL.SPGData
Imports DevExpress.Xpo

Public Class LocationsBLL

    Public Shared Function DeleteLocation(ByVal locationID As Integer, ByVal session As Session) As Boolean

        Dim location As Locations = session.GetObjectByKey(Of Locations)(locationID)

        If location.LocationTransfersFrom.Count > 0 OrElse location.LocationTransfersTo.Count > 0 OrElse location.LocationReceivings.Count > 0 OrElse location.LocationsInventory.Count > 0 OrElse location.LocationShippings.Count > 0 _
                    OrElse location.LocationProduction.Count > 0 OrElse IsDefaultLocation(location) Then
            MessageBox.Show("You can't delete this location there are records associated to it")
        Else
            location.Delete()
        End If

        Return location.IsDeleted

    End Function

    Public Shared Function IsDefaultLocation(ByVal location As Locations) As Boolean

        If CompanySettingsBLL.GetDefaultLocation() IsNot Nothing Then
            Return location.Oid = CompanySettingsBLL.GetDefaultLocation().Oid
        End If

        Return False


    End Function

    Public Shared Function ValidateRecord(ByVal locationCode As String) As Boolean

        If String.IsNullOrEmpty(locationCode) Then
            Return False
        End If

        Return True

    End Function

End Class

File: SuperiorPackGroup/BLL/LocationsBLL.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;

using DXDAL.SPGData;
using DevExpress.Xpo;

namespace SuperiorPackGroup
{
	public class LocationsBLL
	{

		public static bool DeleteLocation(int locationID, Session session)
		{

			Locations location = session.GetObjectByKey<Locations>(locationID);

			if (location.LocationTransfersFrom.Count > 0 || location.LocationTransfersTo.Count > 0 || location.LocationReceivings.Count > 0 || location.LocationsInventory.Count > 0 || location.LocationShippings.Count > 0 || location.LocationProduction.Count > 0 || IsDefaultLocation(location))
			{
				MessageBox.Show("You can't delete this location there are records associated to it");
			}
			else
			{
				location.Delete();
			}

			return location.IsDeleted;

		}

		public static bool IsDefaultLocation(Locations location)
		{

			if (CompanySettingsBLL.GetDefaultLocation() != null)
			{
				return location.Oid == CompanySettingsBLL.GetDefaultLocation().Oid;
			}

			return false;


		}

		public static bool ValidateRecord(string locationCode)
		{

			if (string.IsNullOrEmpty(locationCode))
			{
				return false;
			}

			return true;

		}

	}

}